data/transactions/verify/txn.go

```diff
@@ -225,6 +226,18 @@ func txnGroupBatchPrep(stxs []transactions.SignedTxn, contextHdr *bookkeeping.Bl
 			minFeeCount++
 		}
 		feesPaid = basics.AddSaturate(feesPaid, stxn.Txn.Fee.Raw)
+		lSigPooledSize += stxn.Lsig.Len()
+	}
+	if groupCtx.consensusParams.EnableLogicSigSizePooling {
+		lSigMaxPooledSize := len(stxs) * int(groupCtx.consensusParams.LogicSigMaxSize)
+		if lSigPooledSize > lSigMaxPooledSize {
+			errorMsg := fmt.Sprintf(
+				"txgroup had %d bytes of LogicSigs, more than the available pool of %d bytes",
+				lSigPooledSize, lSigMaxPooledSize,
+			)
+			err = &TxGroupError{err: errors.New(errorMsg), GroupIndex: -1, Reason: TxGroupErrorReasonLogicSigFailed}
```

Comment:
We need a different `Reason` here, since this counter seems designed to count how often LogicSigs _execute_ but fail.  I suppose maybe `TxGroupErrorReasonNotWellFormed` would be ok, rather than introduce a new one.

Reply:
I agree, I also changed this [comment](https://github.com/algorand/go-algorand/pull/6057/files#diff-d3741bc12fa9faf0a148f027517dbfde4942668add6ee4ac7b4032543db4b0f5R96) for consistency, since `txn.WellFormed` does not fail on this